sybil/parsers/rest/capture.py

```diff
@@ -28,14 +28,14 @@ def indent_matches(line, indent):
     return False
 
 
-class DocumentReverseIterator(list):
+class DocumentReverseIterator(List[Any]):
```

Comment:
I think it's a `List[str]`, does that not work?

Reply:
It does not.

Iterating over a `List[str]` produces `str`s. Iterating over this produces `Tuple[int, str]`s.

Changing this to `List[str]` gives us the following `mypy` error:

```
sybil/parsers/rest/capture.py:38: error: Return type "Iterator[tuple[int, str]]" of "__iter__" incompatible with return type "Iterator[str]" in supertype "list"  [override]
```

Reply:
What happens with `List[tuple[int, str]]`?

Reply:
> What happens with List[tuple[int, str]]?

If we try `List[tuple[int, str]]` we are telling the type checker that:

`line = self[self.current_line]` would make `line` a `tuple[int, str]`.
Instead, `line` is a `str`.

If you'd like to avoid `List[Any]`, then my suggestion is rename `__iter__` to something like `iter_lines`.

How would you like to proceed?

Reply:
Precisely the error given is:

```
$ mypy sybil/parsers/rest/capture.py
sybil/parsers/rest/capture.py:34: error: Incompatible types in assignment (expression has type "list[str]", target has type "Iterable[tuple[int, str]]")  [assignment]
sybil/parsers/rest/capture.py:43: error: Incompatible types in "yield" (actual type "tuple[int, tuple[int, str]]", expected type "tuple[int, str]")  [misc]
sybil/parsers/rest/capture.py:79: error: Argument 1 to "join" of "str" has incompatible type "list[tuple[int, str]]"; expected "Iterable[str]"  [arg-type]
```

Reply:
I remember now... this is a particularly fiddly little class. I've refactored and sorted out the issues you raise above in 9b51b0547c7ae5464407120dde712d928caa4499